io/pci/EEH.py

```diff
@@ -134,6 +158,22 @@ def test_eeh_basic_pe(self):
                                 break
                             else:
                                 self.log.info("PE recovered successfully")
+                    time.sleep(10)
+                    if (self.pci_class_name == 'scsi_host' and
+                            self.max_freeze >= num_of_hit):
+                        paths_after_eeh = self.get_paths(self.pci_device)
+                        host_state_after_eeh = pci.get_port_state(
+                            self.pci_device)
+                        self.log.info("Paths after EEH: %s",
+                                      paths_after_eeh)
+                        self.log.info("hosts state after EEH: %s",
+                                      host_state_after_eeh)
+                        if paths_before_eeh != paths_after_eeh:
+                            self.fail("All Paths are failed to recover"
+                                      " after EEH")
+                        if host_state_before_eeh != host_state_after_eeh:
+                            self.fail("Hosts state differ after EEH")
+
```

Comment:
@maramsmurthy where are we checking the states of the paths here.
that is the major functionality check here what we need to incorporate.
follow below steps while doing validation,
1. avoid static sleep times, and use wait_for()
2.  after each eeh hit, you should check each paths_status== online, and fc_host_state == online.

what if you have the adapter already in faulty state?
what if if you have some eeh hit before u try to inject error and few paths are already in failed state?
you will be still working on a fault adapter and making the test case PASS.

Reply:
please refer, port_bounce or pci_hotplug test script.
or let me know I can help you out to add the validation part.

Reply:
Addressing it in PR "https://github.com/avocado-framework-tests/avocado-misc-tests/pull/2590". New validation added for every path.